src/test/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/resource/UserResourceIntTest.kt

```diff
@@ -160,6 +160,85 @@ class UserResourceIntTest : IntegrationTestBase() {
     }
   }
 
+  @DisplayName("GET /users/user?email={emailAddress}")
+  @Nested
+  inner class GetUserByEmailAddress {
+    private val matchByUserName = "$[?(@.username == '%s')]"
+
+    @BeforeEach
+    internal fun createUsers() {
+      with(dataBuilder) {
+        generalUser().username("marco.rossi").firstName("Marco").lastName("Rossi")
+          .email("marco@justice.gov.uk")
+          .atPrison("WWI")
+          .buildAndSave()
+      }
+    }
+
+    @AfterEach
+    internal fun deleteUsers() = dataBuilder.deleteAllUsers()
+
+    @Test
+    fun `access forbidden when no authority`() {
+      webTestClient.get().uri {
+        it.path("/users/user")
+          .queryParam("email", "marco@justice.gov.uk").build()
+      }
+        .exchange()
+        .expectStatus().isUnauthorized
+    }
+
+    @Test
+    fun `access forbidden when no role`() {
+
+      webTestClient.get().uri {
+        it.path("/users/user")
+          .queryParam("email", "marco@justice.gov.uk").build()
+      }
+        .headers(setAuthorisation(roles = listOf()))
+        .exchange()
+        .expectStatus().isForbidden
+    }
+
+    @Test
+    fun `get user forbidden with wrong role`() {
+
+      webTestClient.get().uri {
+        it.path("/users/user")
+          .queryParam("email", "marco@justice.gov.uk").build()
+      }
+        .headers(setAuthorisation(roles = listOf("ROLE_BANANAS")))
+        .exchange()
+        .expectStatus().isForbidden
+    }
+
+    @Test
+    fun `get user when email address not found`() {
+      webTestClient.get().uri {
+        it.path("/users/user")
+          .queryParam("email", "missing@justice.gov.uk").build()
+      }
+        .headers(setAuthorisation(roles = listOf("ROLE_MANAGE_NOMIS_USER_ACCOUNT")))
+        .exchange()
+        .expectStatus().isOk
+        .expectBody().json("[]")
+    }
+
+    @Test
+    fun `get user by email`() {
+      webTestClient.get().uri {
+        it.path("/users/user")
+          .queryParam("email", "marco@justice.gov.uk").build()
+      }
+        .headers(setAuthorisation(roles = listOf("ROLE_MANAGE_NOMIS_USER_ACCOUNT")))
+        .exchange()
+        .expectStatus().isOk
+        .expectBody()
+        .jsonPath("$").isArray
+        .jsonPath(matchByUserName, "marco.rossi").exists()
+    }
+  }
+
```

Comment:
Maybe add a test where 2 users share the same email address

Reply:
yes - good idea - will add in next PR